src/main/java/com/sap/cloudfoundry/client/facade/adapters/RawCloudServiceOffering.java

```diff
@@ -14,28 +13,31 @@
 @Value.Immutable
 public abstract class RawCloudServiceOffering extends RawCloudEntity<CloudServiceOffering> {
 
-    public abstract Resource<ServiceEntity> getResource();
+    public abstract ServiceOfferingResource getResource();
 
     public abstract List<Derivable<CloudServicePlan>> getServicePlans();
 
     @Override
     public CloudServiceOffering derive() {
-        Resource<ServiceEntity> resource = getResource();
-        ServiceEntity entity = resource.getEntity();
+        ServiceOfferingResource resource = getResource();
         return ImmutableCloudServiceOffering.builder()
                                             .metadata(parseResourceMetadata(resource))
-                                            .name(entity.getLabel())
-                                            .isActive(entity.getActive())
-                                            .isBindable(entity.getBindable())
-                                            .description(entity.getDescription())
-                                            .extra(entity.getExtra())
-                                            .docUrl(entity.getDocumentationUrl())
-                                            .infoUrl(entity.getInfoUrl())
-                                            .version(entity.getVersion())
-                                            .provider(entity.getProvider())
-                                            .brokerName(entity.getServiceBrokerName())
-                                            .uniqueId(entity.getUniqueId())
-                                            .url(entity.getUrl())
+                                            .name(resource.getName())
+                                            .isActive(resource.getAvailable())
+                                            .isBindable(resource.getBrokerCatalog()
+                                                                .getFeatures()
+                                                                .getBindable())
```

Comment:
Could these methods return a null object?

Reply:
I didn't find null field for broker catalog when execute `GET v3/service_offerings` . In documentation I don't find anything described that it is possible to be null.